File: packages/device/android/src/main/java/io/capkit/device/error/ErrorMessages.kt
package io.capkit.device.error

/**
 * Canonical error messages shared across platforms.
 * Keep these strings identical on iOS and Android.
 */
object ErrorMessages {
  const val UNAVAILABLE = "Feature is unavailable on this device or configuration."
  const val CANCELLED = "Operation was cancelled."
  const val PERMISSION_DENIED = "Required permission is denied or not granted."
  const val INIT_FAILED = "Native initialization failed."
  const val INVALID_INPUT = "Invalid or missing input parameters."
  const val UNKNOWN_TYPE = "Unsupported or invalid input type."
  const val NOT_FOUND = "Requested resource not found."
  const val CONFLICT = "Operation conflict with current state."
  const val TIMEOUT = "Operation timed out."
  const val INTERNAL_ERROR = "Internal error."
  const val UNEXPECTED_NATIVE_ERROR = "Unexpected native error in device operation."
}

File: packages/device/ios/Sources/DevicePlugin/error/ErrorMessages.swift
import Foundation

/**
 Canonical error messages shared across platforms.

 These strings should remain byte-identical on iOS and Android
 whenever they represent the same failure condition.
 */
enum ErrorMessages {
    static let unavailable = "Feature is unavailable on this device or configuration."
    static let cancelled = "Operation was cancelled."
    static let permissionDenied = "Required permission is denied or not granted."
    static let initFailed = "Native initialization failed."
    static let invalidInput = "Invalid or missing input parameters."
    static let unknownType = "Unsupported or invalid input type."
    static let notFound = "Requested resource not found."
    static let conflict = "Operation conflict with current state."
    static let timeout = "Operation timed out."
    static let internalError = "Internal error."
    static let unexpectedNativeError = "Unexpected native error in device operation."
}
